### DISCUSSION TITLE
Whats a good way to pass data to the page-elements?

### DISCUSSION THREAD

**Discussion by Alos** [IBM/pwa-lit-template#331] [Q&A]
Because the vaadin-router dynamically replaces the page-elements shown inside the `<main>` tag, there is no chance for the `app-index` to pass some properties down to the page-elements. 

Whats the best way to achieve this?

**Comment by Tansito** [comment]
Thanks for your question @Alos! When we want to pass information between pages we usually use a library to manage the state of the application. Examples of those libraries can be `redux` or `apollo-client` if you use GraphQL.

Some time ago we were using `redux` to manage the state of some apps (in [Vaadin](https://vaadin.com/learn/tutorials/lit-element) you have a good tutorial to configure it, for example) but nowadays we are using a lot [Apollo GraphQL](https://www.apollographql.com/) through [Apollo Elements](https://apolloelements.dev/) for Lit and the experience using it is being quite graceful.
